Automattic/wp-calypso
# Stats Widget: Introduce new Stats Widget with a basic chart

Related to #75102 

## Proposed Changes

* Add `widget/index.jsx` as the Stats Widget application root.
* Add `widget/mini-chart.jsx`
* Update existing chart-related components to fit the design.

The `widget-loader` is about `~9KB`, and the main load is `~310KB` atm. The `wpcom` client takes up a huge place (50KB+) there, which will be replaced with a simple wrapper in a follow up PR.

<img width="926" alt="截圖 2023-04-01 上午4 25 43" src="https://user-images.githubusercontent.com/6869813/229223573-f1a62b99-c801-4583-a554-55ec294cd4f6.png">

## Testing Instructions



* Open your local Jetpack site.
* Check out the branch of the Jetpack PR: https://github.com/Automattic/jetpack/pull/29929/.
* Build Jetpack if necessary: `jetpack build plugins/jetpack`.
* Run `STATS_PACKAGE_PATH=/path/to/jetpack/projects/packages/stats-admin yarn dev` under `calypso/apps/odyssey-stats`.
* Navigate to `/wp-admin/index.php?odyssey_widget=1`.
* Ensure the Odyssey Widget displays in line with the design.
* Ensure i18n works
* Ensure the chart respect your site timezone
* Ensure periods in Calypso Stats still works okay
* Click the bars
* Ensure it takes you to correct Odyssey Stats with correct params

<img width="540" alt="image" src="https://user-images.githubusercontent.com/1425433/229943360-2ac70779-3ff2-42b3-90aa-78849b51b3cd.png">


## Pre-merge Checklist
